make test fails: test_metadata_normalization

Hi!
I'm trying to build vdirsyncer for Arch Linux to move it to [community].
However, the `make test` fails with the below message:

```
=================================== FAILURES ===================================
________________ TestMemoryStorage.test_metadata_normalization _________________
tests/storage/__init__.py:301: in test_metadata_normalization
    st.none(),
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 2 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
---------------------------------- Hypothesis ----------------------------------
You can add @seed(4863082389373184927620322932099709995) to this test or run pytest with --hypothesis-seed=4863082389373184927620322932099709995 to reproduce this failure.
============== 1 failed, 370 passed, 115 skipped in 22.99 seconds ==============
make: *** [Makefile:59: test] Error 1
```
My setup is:

* vdirsyncer 0.16.3 (attempted)
* python 3.6.4

With `make DETERMINISTIC_TESTS=true test` it lead to the same problem:

```
=================================== FAILURES ===================================
______________ TestFilesystemStorage.test_metadata_normalization _______________
tests/storage/__init__.py:301: in test_metadata_normalization
    st.none(),
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 5 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
________________ TestMemoryStorage.test_metadata_normalization _________________
tests/storage/__init__.py:301: in test_metadata_normalization
    st.none(),
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 5 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
```
So in the meantime I'm patching the test out to build the package.

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,7 @@ build
 env
 *.egg-info
 .cache
+.pytest_cache
 .eggs
 .egg
 .xprocess

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,7 +2,8 @@
   "branches": {
     "only": [
       "auto",
-      "master"
+      "master",
+      "/^.*-maintenance$/"
     ]
   },
   "cache": "pip",

--- a/.travis.yml
+++ b/.travis.yml
@@ -90,28 +90,6 @@
         "python": "3.6"
       },
       {
-        "env": "BUILD=test DAV_SERVER=owncloud REQUIREMENTS=release ",
-        "python": "3.6"
-      },
-      {
-        "env": "BUILD=test DAV_SERVER=nextcloud REQUIREMENTS=release ",
-        "python": "3.6"
-      },
-      {
-        "env": "BUILD=test DAV_SERVER=baikal REQUIREMENTS=release ",
-        "python": "3.6"
-      },
-      {
-        "env": "BUILD=test DAV_SERVER=davical REQUIREMENTS=release ",
-        "if": "NOT (type IN (pull_request))",
-        "python": "3.6"
-      },
-      {
-        "env": "BUILD=test DAV_SERVER=icloud REQUIREMENTS=release ",
-        "if": "NOT (type IN (pull_request))",
-        "python": "3.6"
-      },
-      {
         "env": "BUILD=test DAV_SERVER=fastmail REQUIREMENTS=release ",
         "if": "NOT (type IN (pull_request))",
         "python": "3.6"

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -9,6 +9,13 @@ Package maintainers and users who have to manually update their installation
 may want to subscribe to `GitHub's tag feed
 <https://github.com/pimutils/vdirsyncer/tags.atom>`_.
 
+Version 0.16.4
+==============
+
+*released on 05 February 2018*
+
+- Fix tests for new Hypothesis version. (Literally no other change included)
+
 Version 0.16.3
 ==============
 

--- a/Makefile
+++ b/Makefile
@@ -72,7 +72,7 @@ install-servers:
 	done
 
 install-test: install-servers
-	pip install -r test-requirements.txt
+	pip install -Ur test-requirements.txt
 	set -xe && if [ "$$REQUIREMENTS" = "devel" ]; then \
 		pip install -U --force-reinstall \
 			git+https://github.com/DRMacIver/hypothesis \
@@ -82,7 +82,7 @@ install-test: install-servers
 	[ -z "$(TEST_EXTRA_PACKAGES)" ] || pip install $(TEST_EXTRA_PACKAGES)
 
 install-style: install-docs
-	pip install flake8 flake8-import-order 'flake8-bugbear>=17.3.0' autopep8
+	pip install -U flake8 flake8-import-order 'flake8-bugbear>=17.3.0' autopep8
 	
 style:
 	flake8
@@ -95,7 +95,7 @@ travis-conf:
 	python3 scripts/make_travisconf.py > .travis.yml
 
 install-docs:
-	pip install -r docs-requirements.txt
+	pip install -Ur docs-requirements.txt
 
 docs:
 	cd docs && make html
@@ -115,7 +115,7 @@ release-deb:
 
 install-dev:
 	pip install -e .
-	[ "$(ETESYNC_TESTS)" = "false" ] || pip install -e .[etesync]
+	[ "$(ETESYNC_TESTS)" = "false" ] || pip install -Ue .[etesync]
 	set -xe && if [ "$(REQUIREMENTS)" = "devel" ]; then \
 	    pip install -U --force-reinstall \
 			git+https://github.com/mitsuhiko/click \

--- a/scripts/make_travisconf.py
+++ b/scripts/make_travisconf.py
@@ -17,7 +17,7 @@ cfg['git'] = {
 }
 
 cfg['branches'] = {
-    'only': ['auto', 'master']
+    'only': ['auto', 'master', '/^.*-maintenance$/']
 }
 
 cfg['install'] = """

--- a/scripts/make_travisconf.py
+++ b/scripts/make_travisconf.py
@@ -44,8 +44,7 @@ for python, requirements in itertools.product(python_versions,
     dav_servers = ("radicale", "xandikos")
 
     if python == latest_python and requirements == "release":
-        dav_servers += ("owncloud", "nextcloud", "baikal", "davical", "icloud",
-                        "fastmail")
+        dav_servers += ("fastmail",)
 
     for dav_server in dav_servers:
         job = {

--- a/setup.cfg
+++ b/setup.cfg
@@ -9,5 +9,5 @@ addopts = --tb=short
 # E731: Use a def instead of lambda expr
 ignore = E731
 select = C,E,F,W,B,B9
-exclude = tests/storage/servers/owncloud/, tests/storage/servers/nextcloud/, tests/storage/servers/baikal/, build/
+exclude = .eggs, tests/storage/servers/owncloud/, tests/storage/servers/nextcloud/, tests/storage/servers/baikal/, build/
 application-package-names = tests,vdirsyncer

--- a/setup.cfg
+++ b/setup.cfg
@@ -7,7 +7,8 @@ addopts = --tb=short
 
 [flake8]
 # E731: Use a def instead of lambda expr
-ignore = E731
+# E743: Ambiguous function definition
+ignore = E731, E743
 select = C,E,F,W,B,B9
 exclude = .eggs, tests/storage/servers/owncloud/, tests/storage/servers/nextcloud/, tests/storage/servers/baikal/, build/
 application-package-names = tests,vdirsyncer

--- a/tests/storage/__init__.py
+++ b/tests/storage/__init__.py
@@ -299,7 +299,7 @@ class StorageTests(object):
 
     @given(value=st.one_of(
         st.none(),
-        printable_characters_strategy.filter(lambda x: x.strip() != x)
+        printable_characters_strategy
     ))
     def test_metadata_normalization(self, requires_metadata, s, value):
         x = s.get_meta('displayname')

